akka-stream/src/main/scala/akka/stream/impl/io/FileSubscriber.scala

```diff
@@ -28,17 +32,13 @@ private[akka] class FileSubscriber(f: File, bytesWrittenPromise: Promise[Long],
 
   override protected val requestStrategy = WatermarkRequestStrategy(highWatermark = bufSize)
 
-  private var raf: RandomAccessFile = _
   private var chan: FileChannel = _
 
   private var bytesWritten: Long = 0
 
   override def preStart(): Unit = try {
-    raf = new RandomAccessFile(f, "rw") // best way to express this in JDK6, OpenOption are available since JDK7
-    chan = raf.getChannel
-
-    // manually supporting appending to files - in Java 7 we could use OpenModes: FileChannel.open(f, openOptions.asJava)
```

Comment:
Yeah, we can make the options fix as part of that ticket (and then allow making the offset somehow).
I'm not sure, but APPEND implies unix file permission WRITE so I guess it may not be needed.